index.js

```diff
@@ -177,16 +166,16 @@ export default {
   events: EventEmitter,
   removeListeners() {
     if (onPaymentProvideListener) {
-      EventEmitter.removeListener(onPaymentProvideListener);
+      onPaymentProvideListener.remove();
```

Comment:
I see what you're trying to do, but basically here you could stay with many listeners still open as you're changing the variables declared on top.

Reply:
I don't think so. The map is removed, so you will have only one listener at a time, given that you properly call `removeListeners` at `componentWillUnmount`, for example.

Reply:
The top `let` variables are each time assigned a new `undefined` value when the listener is attached. Hence the `removeListeners` isn't doing anything and more and more listeners are registered.